pkg/controller/stacks/stack/stack.go

```diff
@@ -116,6 +121,13 @@ func Setup(mgr ctrl.Manager, l logging.Logger, hostControllerNamespace string) e
 		log:          l.WithValues("controller", name),
 	}
 
+	if err := mgr.GetFieldIndexer().IndexField(&apiextensions.CustomResourceDefinition{}, crdAPIGroupField, func(rawObj runtime.Object) []string {
```

Comment:
It may be possible to create a pseudo-index (like a SQL composite index) here which could be used to avoid the extra field filters happening when we do use the MatchingFields query. I'm not sure if that is possible though and I kept this simple for now.

Reply:
Adding `//	Use of MatchingFields on fields not indexed by the API requires a cache index`